app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1167,6 +1167,118 @@ export default Controller.extend(bufferedProperty("model"), {
     }
   },
 
+  _togglePostBookmark(post) {
+    let postEl = document.querySelector(`[data-post-id="${post.id}"]`);
```

Comment:
This is still not ideal unfortunately. Could we move the `querySelector` part into a component? The rest seems OK.

Reply:
I moved it to bookmark modal controller so it is closer to a place where that data is needed. Do you think it is fine? Or you prefer to create a separate component - something like `BookmarkLocalDateOption` which would be called in the `bookmark.hbs` modal template

Reply:
All DOM manipulation should be done in components, yes. The modal controller is slightly better but long term we don't want any DOM stuff done in any controllers.

Reply:
Introduced `bookmark-local-date` component :)